main/src/cgeo/geocaching/models/WaypointParser.java

```diff
@@ -0,0 +1,454 @@
+package cgeo.geocaching.models;
+
+import cgeo.geocaching.calculator.CoordinatesCalculateUtils;
+import cgeo.geocaching.calculator.FormulaParser;
+import cgeo.geocaching.calculator.VariableData;
+import cgeo.geocaching.enumerations.WaypointType;
+import cgeo.geocaching.location.Geopoint;
+import cgeo.geocaching.location.GeopointFormatter;
+import cgeo.geocaching.location.GeopointParser;
+import cgeo.geocaching.location.GeopointWrapper;
+import cgeo.geocaching.settings.Settings;
+import cgeo.geocaching.utils.TextUtils;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.jetbrains.annotations.NotNull;
+
+public class WaypointParser {
```

Comment:
Kudos for extraction of the parsing methods into an own class! :-)

Reply:
As reminder for me:
check, if there are more changes in parsing methods in master which has to be adapted to WaypointParser